features/lege-waarden.feature

```diff
@@ -156,7 +215,27 @@ Functionaliteit: Leveren van lege waarden
       Dan heeft de response een persoon met een 'kind' met een leeg 'naam' object
       # "kinderen": [{ "naam": {} }]
```

Comment:
Er zit voor mijn gevoel toch enige inconsistentie in de wijze waarop we omgaan met het lege properties. Als we de partner opvragen van een persoon die nog nooit een partner heeft gehad (fields: partners) dan is het antwoord:

    "personen": [ { "partners": [] } ]

Zie regel 141.
Je neemt dat wat je opvraagt wel op in het antwoord maar leeg.

Als we echter het voorvoegsel opvragen van de naam van een kind (fields: kinderen.naam.voorvoegsel) dan is het antwoord:

    "kinderen": [{ "naam": {} }]

Je neemt dat wat je opvraagt nu niet op in het antwoord.
Als je consistent zou zijn zou je zeggen dat het antwoord als volgt zou moeten zijn:

    "kinderen": [{ "naam": { "voorvoegsel": { } } }]

Reply:
behalve dat voorvoegsel geen object is. Dat zou dan een null object zijn
`"kinderen": [{ "naam": { "voorvoegsel": null } }]`

We laten nu consequent velden die niet bestaan (null, none of undefined) weg en leveren objecten die wel bestaan (waarde hebben), maar geen van de gevraagde velden heeft een waarde, als leeg object.

Wanneer er geen kinderen zijn, moet een lege array geleverd moet worden. Voor het uitlezen van gegevens is dat makkelijker dan niet leveren, maar de consequente redenatie zou zijn dat we iets dat niet bestaat niet leveren als iets leegs. We leveren als er geen kinderen zijn "kinderen": []. Volgens mij hebben we hier gekozen voor gemak in gebruik boven consequent.

In geval alleen de geslachtsnaam gevuld is met "." (standaard/onbekendwaarde) en de andere naamvelden bestaan niet, leveren we "naam": {}. Volgens mij hebben we ook dat gedaan om het uitlezen van naamvelden makkelijker te maken (gebruiker zou dan zekerder zijn dat het hele pad tot het gevraagde veld in ieder geval bestaat). Of dat helemaal waar is voor andere velden dan naam betwijfel ik. Bijvoorbeeld kinderen.geboorte of partners. aangaanHuwelijkPartnerschap kunnen ook helemaal niet bestaan, zodat jsonPath kinderen[0].geboorte.datum alsnog een fout zoals trying to get property of null object kan geven.
@MelvLee hoe zinvol is dit? En is het wel terecht om "naam": {} te leveren als geslachtsnaam=".", maar geen naam te leveren als ook geslachtsnaam ontbreekt?
> Rule: Een object (groep) wordt geleverd als leeg object wanneer er ten minste één gegeven in die groep bestaat, maar geen enkel veld in die groep geleverd moet worden
> - hierbij geldt ook een veld met een standaardwaarde, leegwaarde of boolean false voor het bestaan van de groep